### PR TITLE
SIMSBIOHUB-150: Database Modelling to Support 'Groups'

### PR DESCRIPTION
**Pull request by NickPhura**
## Links to Jira Tickets

https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-150

## Description of Changes

- Add migration to create project group tables (7 new tables)
  - Should all be net new tables, with no impact on any existing tables

- Removed old team members from user seed
- Added Macgregor to user seed

## Testing Notes

- Migration runs and tables are created successfully

### CODE REVIEW

File: database/src/migrations/20230725000000_project_groups.ts [hunk 1]
```diff
@@ -0,0 +1,361 @@
...
+    COMMENT ON COLUMN grouping_permission.create_date              IS 'The datetime the record was created.';
+    COMMENT ON COLUMN grouping_permission.create_user              IS 'The id of the user who created the record as identified in the system user table.';
+    COMMENT ON COLUMN grouping_permission.update_date              IS 'The datetime the record was updated.';
+    COMMENT ON COLUMN grouping_permission.update_user              IS 'The id of the user who updated the record as identified in the system user table.';
+    COMMENT ON COLUMN grouping_permission.revision_count           IS 'Revision count used for concurrency control.';
+    COMMENT ON TABLE  grouping_permission                          IS 'Grouping permissions.';
+
+    ----------------------------------------------------------------------------------------
+
+    CREATE TABLE grouping_project(
```
Comment (al-rosenthal) [inline database/src/migrations/20230725000000_project_groups.ts]: other join tables on projects are `project_entity` so `project_group` or `project_region` ect.

Comment (NickPhura) [reply database/src/migrations/20230725000000_project_groups.ts]: Discussed, no change needed atm.

**Review by al-rosenthal — DISMISSED** [review]
🍦

**Review by al-rosenthal — APPROVED** [review]
🥓

**Pull request merged**